src/libraries/Microsoft.PowerFx.Core/Functions/UserDefinedFunction.cs

```diff
@@ -200,14 +206,15 @@ public static IEnumerable<UserDefinedFunction> CreateFunctions(IEnumerable<UDF>
 
             var userDefinedFunctions = new List<UserDefinedFunction>();
             var texlFunctionSet = new TexlFunctionSet();
-            errors = new List<TexlError>();
+            errors = new List<TexlError>();
 
             foreach (var udf in uDFs)
             {
                 Contracts.Assert(udf.IsParseValid);
 
                 var udfName = udf.Ident.Name;
-                if (_restrictedUDFNames.Contains(udfName) || texlFunctionSet.AnyWithName(udfName) || BuiltinFunctionsCore._library.AnyWithName(udfName) || BuiltinFunctionsCore.OtherKnownFunctions.Contains(udfName))
+                if (_restrictedUDFNames.Contains(udfName) || texlFunctionSet.AnyWithName(udfName) ||
+                    nameResolver.Functions.WithName(udfName).Any(func => func.IsRestrictedUDFName))
                 {
                     errors.Add(new TexlError(udf.Ident, DocumentErrorSeverity.Severe, TexlStrings.ErrUDF_FunctionAlreadyDefined, udfName));
```

Comment:
>ErrUDF_FunctionAlreadyDefined [](http://example.com/codeflow?start=98&length=29)

In a host such as CLI or MCS which doesn't have the concept of UpdateContext or Navigate, will the error message "Function {0} is already defined." make sense to them?

Reply:
Added a separate error for restricted names